src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/WrapWithTag/WrapWithTagEndpoint.cs

```diff
@@ -19,47 +20,38 @@ internal class WrapWithTagEndpoint : IWrapWithTagEndpoint
     {
         private readonly ClientNotifierServiceBase _languageServer;
         private readonly RazorDocumentMappingService _razorDocumentMappingService;
-        private readonly DocumentContextFactory _documentContextFactory;
-        private readonly ILogger _logger;
 
         public WrapWithTagEndpoint(
             ClientNotifierServiceBase languageServer,
-            DocumentContextFactory documentContextFactory,
-            RazorDocumentMappingService razorDocumentMappingService,
-            ILoggerFactory loggerFactory)
+            RazorDocumentMappingService razorDocumentMappingService)
         {
             if (languageServer is null)
             {
                 throw new ArgumentNullException(nameof(languageServer));
             }
 
-            if (documentContextFactory is null)
-            {
-                throw new ArgumentNullException(nameof(documentContextFactory));
-            }
-
             if (razorDocumentMappingService is null)
             {
                 throw new ArgumentNullException(nameof(razorDocumentMappingService));
             }
 
-            if (loggerFactory is null)
-            {
-                throw new ArgumentNullException(nameof(loggerFactory));
-            }
-
             _languageServer = languageServer;
-            _documentContextFactory = documentContextFactory;
             _razorDocumentMappingService = razorDocumentMappingService;
-            _logger = loggerFactory.CreateLogger<WrapWithTagEndpoint>();
         }
 
-        public async Task<WrapWithTagResponse?> Handle(WrapWithTagParamsBridge request, CancellationToken cancellationToken)
+        public bool MutatesSolutionState => false;
+
+        public TextDocumentIdentifier GetTextDocumentIdentifier(WrapWithTagParams request)
+        {
+            return request.TextDocument;
```

Comment:
Is there ever a time when this won't be a simple property accessor? Can we make this cleaner somehow?

Reply:
There will always be A simple property accessor, but in some cases it's a TextDocumentIdentifier, in some cases it's a URI, sometimes it's a `TextDocumentItem`. There was a lot of talk about simplifying this on the PR for CLaSP in Roslyn and none of the ideas felt like a clear improvement to me. On the more specific area of Razor a follow-up PR might  be able to paper over this using an abstract class most of the endpoints inherit from, but that's likely to be messy and have a lot of exceptions.